lib/rules/method.js

```diff
@@ -99,6 +105,8 @@ function checkCallExpression(ruleHelper, callExpr, node) {
     case "CallExpression":
     case "ThisExpression":
     case "NewExpression":
+    case "TSAsExpression":
+    case "TSTypeAssertion":
```

Comment:
Same as above?
This could be bad: `foo.innerHTML = (<string>code);`

I think you will need a utility function that decides whether a specific type needs a check with an allow-list of safe types that don't (numbers, booleans, literals, what else?).
Eventually, we'll want that list to be customized through rule configs, but you can flag that as an issue for later.

Reply:
Example was bad, like above. We'd still want to look into the thing that's getting a type assertion though.

Reply:
I think I handled this - let me know how it looks!

Reply:
reference link:
https://github.com/mozilla/eslint-plugin-no-unsanitized/pull/137/files#diff-69b92fc433cd9abc7aff2c0c2fd0e97aR77